docs/overview.rst

```diff
@@ -181,23 +176,32 @@ or HPy Hybrid ABIs require installing the ``hpy.universal`` package on
 CPython and have no further requirements on Pythons that support HPy natively.
 
 
-C extensions
-------------
+Benefits for the Python ecosystem
+---------------------------------
 
-If you are writing a Python extension in C, you are a consumer of the HPy
-API. There are three big advantages in using HPy instead of the old Python/C
-API:
-
-  - Speed on PyPy, GraalPython and other alternative implementations: according
-    to early :ref:`benchmarks`, an extension written in HPy can be ~3x faster
-    than the equivalent extension written in Python/C.
+The HPy project offers some benefits to the python ecosystem, both to Python
+users and to library developers.
 
+  - C extensions can achieve much better speed on alternative implementions,
+    including PyPy and GraalPython: according to early :ref:`benchmarks`, an
+    extension written in HPy can be ~3x faster than the equivalent extension
+    written in Python/C.
   - Improved debugging: when you load extensions in :ref:`debug-mode:debug mode`,
     many common mistakes are checked and reported automatically.
-
-  - Universal binaries: you can choose to distribute only Universal ABI
-    binaries. This comes with a small speed penalty on CPython, but for
-    non-performance critical libraries it might still be a good tradeoff.
+  - Universal binaries: libraries can choose to distribute only Universal ABI
+    binaries. By doing so, they can support all versions of CPython for which
+    an hpy loader exists, including versions of CPython that have not yet been
+    developed. This currently comes with a small speed penalty on CPython, but
+    for non-performance critical libraries it might still be a good tradeoff.
+  - Implementation-switching Python environments: Users can currently create
+    Python environments for a given implementation with a set of library
+    requirements. With universal binaries, users can create Python environments
+    with any permutation of both library requirements and choice of
+    implementation. This allows users to choose the best implementation for
+    their workload.
```

Comment:
I needed few seconds to parse this. It also follows from the previous point, right?, but maybe it can be viewed as such an advantage that it is worth pointing out explicitly. From my POV it seems bit complex, too abstract and obscure here, but I am open to be convinced otherwise

Reply:
Elsewhere in the documentation we highlight the advantages to Python implementations. Before my change here - the advantages discussed are for authors of CPython, which follows from the old title of this section.

I renamed the title and added this point (and the below one) to show that HPy offers advantages not only to CPython extension authors, but to the ecosystem as a whole.  This includes users.  As a Python user, if I get python from python.org nothing changes.  But if I use a higher-level python environment system (such as conda) - it is possible for a hypothetical python environment system - let's call it honda - to do things like this:

```
honda create-env Variant1 --python pypy:latest --requirements requirements1.txt
honda create-env Variant2 --python CPython:3.12 --requirements requirements2.txt
honda create-env Variant3 --python graal:latest --requirements requirements1.txt
honda create-env Variant4 --python cinder:20 --requirements requirements3.txt
honda create-env Variant5 --python nogil-python:11 --requirements requirements4.txt
honda create-env Variant6 --python gc-nogil-python:22 --requirements requirements2.txt
honda create-env Variant7 --python my-experimental:5 --requirements requirements2.txt
```

where python interpreters use a docker-like convention for name:version.  Then, users could benchmark one against another to try things out. This could be interesting for python users, who could switch implementations easily, letting `honda` figure out where the packages in the requirements file are supported for a given implementation.

Reply:
Thanks for the explanation. I get the idea. I still wonder if this isn't too abstract for an overview, adding an example or being more concrete would help, but then it would be too long/complex topic for an overview imo. Maybe a concrete example how it can help specifically conda users would be more explanatory, but still short? At the same time, it's definitely not a big issue. Unless anyone else also shares my concerns, I would not sink too much time into this. From my POV this is not a blocker.

Reply:
I reworded it.